crates/mirror-mirror/src/try_visit.rs

```diff
@@ -0,0 +1,201 @@
+use crate::{
+    type_info::{OpaqueType, Type, VariantField},
+    Reflect, ScalarRef,
+};
+
+macro_rules! visit_scalar_fn {
+    ($name:ident, $ty:ty) => {
+        #[allow(clippy::ptr_arg)]
+        fn $name(&mut self, value: $ty) -> Result<(), Self::Error> {
+            Ok(())
+        }
+    };
+}
+
+#[allow(unused_variables)]
+pub trait TryVisit {
+    type Error;
+
+    visit_scalar_fn!(try_visit_usize, usize);
+    visit_scalar_fn!(try_visit_u8, u8);
+    visit_scalar_fn!(try_visit_u16, u16);
+    visit_scalar_fn!(try_visit_u32, u32);
+    visit_scalar_fn!(try_visit_u64, u64);
+    visit_scalar_fn!(try_visit_u128, u128);
+    visit_scalar_fn!(try_visit_i8, i8);
+    visit_scalar_fn!(try_visit_i16, i16);
+    visit_scalar_fn!(try_visit_i32, i32);
+    visit_scalar_fn!(try_visit_i64, i64);
+    visit_scalar_fn!(try_visit_i128, i128);
+    visit_scalar_fn!(try_visit_bool, bool);
+    visit_scalar_fn!(try_visit_char, char);
+    visit_scalar_fn!(try_visit_f32, f32);
+    visit_scalar_fn!(try_visit_f64, f64);
+    visit_scalar_fn!(try_visit_string, &String);
+
+    fn try_visit_opaque(
+        &mut self,
+        value: &dyn Reflect,
+        ty: OpaqueType<'_>,
+    ) -> Result<(), Self::Error> {
+        Ok(())
+    }
```

Comment:
I left out methods to visit intermediates like `&dyn Struct`, `&dyn Enum`, etc, because I didn't need that. But probably makes sense to add in the future.

Reply:
Would it make sense to have them implemented as `todo!()`, or even better, a specific error that signals "not implemented yet", so the visitor implementor is aware that this can happen?

Reply:
owait, I realize now the methods just don't exist, so it's known to the implementer. Please ignore me :sweat_smile: